crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -293,24 +296,109 @@ impl PublishFlakeRef {
         match self {
             PublishFlakeRef::Ssh(ssh_ref) => FlakeRef::GitSsh(ssh_ref),
             PublishFlakeRef::Https(https_ref) => FlakeRef::GitHttps(https_ref),
+            PublishFlakeRef::File(file_ref) => FlakeRef::GitPath(file_ref),
         }
     }
-}
 
-impl TryFrom<FlakeRef> for PublishFlakeRef {
-    type Error = ConvertFlakeRefError;
+    async fn from_git_path(
+        file_ref: GitRef<protocol::File>,
+        allow_dirty: bool,
+        accept_upstream: bool,
+        nix: &NixCommandLine,
+    ) -> Result<Self, ConvertFlakeRefError> {
+        let local_metadata = {
+            let command = runix::command::FlakeMetadata {
```

Comment:
Just curious - do you use nix instead of git to check this because nix has a different definition of dirty?

Reply:
I wanted to be sure to use whatever nix considers dirty, fail early if the folder is not a flake and not to create 2-3 more commands in git, probably should do that anyway if we dont want to use nix within the conversion